templating/base_render_options.go

```diff
@@ -123,24 +123,80 @@ func OverrideValuesWithBuildInfo(c1 *Config, c2 *Config, opts *BaseRenderOptions
 	return base, nil
 }
 
-// LoadAndRenderSteps loads a template file and renders it according to an optional values file, --set values,
+// LoadAndRenderBuildSteps loads a template file for build and renders it according to an optional values file, --set values,
+// and base render options.
+func LoadAndRenderBuildSteps(ctx context.Context, template *Template, opts *BaseRenderOptions) (string, error) {
```

Comment:
Can we just have a flag instead - called `renderSecrets bool`.

```suggestion
// LoadAndRenderSteps loads a template file and renders it according to an optional values file, --set values,
// and base render options.
func LoadAndRenderSteps(ctx context.Context, template *Template, opts *BaseRenderOptions, renderSecrets bool) (string, error) {
	// return empty rendered string for an empty template.
	if len(template.GetData()) == 0 {
		return "", nil
	}

	var err error

	config := &Config{}
	if opts.ValuesFile != "" {
		if config, err = LoadConfig(opts.ValuesFile); err != nil {
			return "", err
		}
	} else if opts.Base64EncodedValuesFile != "" {
		if config, err = DecodeConfig(opts.Base64EncodedValuesFile); err != nil {
			return "", err
		}
	}

	setConfig := &Config{}
	if len(opts.TemplateValues) > 0 {
		var rawVals string
		rawVals, err = parseValues(opts.TemplateValues)
		if err != nil {
			return "", err
		}

		setConfig = &Config{RawValue: rawVals, Values: map[string]*Value{}}
	}

	mergedVals, err := OverrideValuesWithBuildInfo(config, setConfig, opts)
	if err != nil {
		return "", fmt.Errorf("failed to override values: %v", err)
	}

	engine := NewEngine()
	
	if renderSecrets {
	    // we will pass nil for the secret resolve override so as to use the default resolve function.
	    secrets, err := renderAndResolveSecrets(ctx, template, engine, nil, opts, mergedVals)
	    if err != nil {
		    return "", fmt.Errorf("failed to resolve secrets in the task with error: %v", err)
	    }
	    // update the secrets collection with resolved secrets.
	    mergedVals["Secrets"] = secrets
	}
	
	rendered, err := engine.Render(template, mergedVals)
	if err != nil {
		return "", fmt.Errorf("error while rendering templates: %v", err)
	}

	if rendered == "" {
		return "", errors.New("rendered template was empty")
	}

	return rendered, nil
}
```

Reply:
@shahzzzam as build and exec validations continue to evolve separately we decided to split the methods. /cc @northtyphoon

Reply:
Okay sounds good